sec-priv-q.md

```diff
@@ -0,0 +1,303 @@
+# Security and Privacy Questionnaire
+
+This contains brief answers to a bunch of standard security and privacy
+questions as it relates to [IPA](./IPA-End-to-End.md).  However, this is a case
+where the answers to these standard questions are largely unsatisfactory.  IPA
+is specifically designed to provide cross-site information in more or less
+direct contravention of accepted practice.
+
+It is best to summarize IPA from the relevant perspective first.  This summary
+brings out important details from the explainer, as well as acting as a short
+primer.
+
+
+## IPA Summary
+
+The proposed IPA API (palindrome, whoa!) provides sites with the ability to
+request an encrypted match key.
+
+A match key is an identifier that is intended to uniquely identify a person,
+though it could - depending on implementation choices in user agents - only
+identify a user agent.
+
+IPA allows any site to set a match key: the idea is that a site can provide a
+match key using knowledge they have that can connect a user to multiple user
+agents.  For example, websites on which people log in using multiple user agents
+could use login credentials.  Sites can request that the user agent use the
+match key provided by any other site.  The user agent provides a fallback value
+if no value has been set, which will enable attribution for things that a user
+does on that device.
+
+An encrypted match key is a secret-shared version of a match key that is
+encrypted toward each of the three entities that form a helper party network.
+
+Provided that two of the three parties that form the helper party network do not
+reveal the values they receive to each other and they faithfully execute the
+defined MPC protocol, the value of the match key is never revealed to any
+entity.
+
+IPA defines an MPC protocol that provides sites with an aggregated report that
+can be used for attribution.  The MPC system chosen guarantees *malicious
+security* in the *two out of three honest majority* setting.  In this setting,
+both privacy for inputs and security against tampering are guaranteed as long as
+two of the three helper parties faithfully executes the protocol.  It takes two
+corrupted or malicious helpers that actively work together to access the input
+data provided about users or to spoil the output of the computation.  Any attack
+by a single helper party can only cause the protocol to abort.
+
+User agent implementers, on behalf of their users, need to select helper parties
+that can be trusted to correctly execute the protocol and to only reveal the
+output of any processing they perform.  In addition to the cryptographic
+protections built into the MPC protocol design, the specification recommends
+that helper parties be bound by contractual terms with user agent
+implementations to execute the protocol correctly.
```

Comment:
Is it worth noting that we have considered adding additional technical mitigations to make collusion more difficult, i.e. defence in depth through making the helpers run the protocol in a TEE?